src/utils/apiResponse.utils.ts

```diff
@@ -0,0 +1,29 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+export type ControllerResponse = {
+  isSuccess: number;
+  statusCode: number;
+  message: string;
+  data?: any;
+  error?: any;
+};
+
+export const APIResponse = {
+  create: (
+    isSuccess: number,
+    statusCode: number,
+    message?: string,
+    data?: any,
+    error?: any,
+  ): ControllerResponse => {
+    const response: ControllerResponse = { isSuccess, statusCode, message };
+    if (error) {
+      response.error = error;
+    } else if (!error && data) {
+      response.data = data;
+    } else {
+      response.data = {};
+    }
```

Comment:
PullProofAI

The `APIResponse.create` function can be simplified by directly assigning `data` or `error` based on conditions, eliminating the need for multiple if-else blocks.
```diff
export const APIResponse = {
  create: (
    isSuccess: number,
    statusCode: number,
    message?: string,
    data?: unknown,
    error?: unknown,
  ): ControllerResponse => {
    const response: ControllerResponse = { isSuccess, statusCode, message };
    
    if (error) {
      response.error = error;
    } else {
      response.data = data || {};
    }
    
    return response;
  },
};
```

### Summary
- Updated the `ControllerResponse` type to use specific types instead of `any` for `data` and `error`.
- Simplified the `APIResponse.create` function to directly assign `data` or `error` based on conditions, improving readability and reducing unnecessary branching logic.


<!-- This is an auto-generated comment by PullProofAI -->